src/components/Stake/StakingMovements.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  GU,
+  IconAttention,
+  IconCheck,
+  IconClock,
+  IconCross,
+  IconLock,
+  formatTokenAmount,
+  Tag,
+  useTheme,
+  DataView,
+} from '@aragon/ui'
+import { dateFormat, toMs } from '../../utils/date-utils'
+import InternalLink from '../InternalLink'
+
+function StakingMovements({ stakingMovements, token }) {
+  const theme = useTheme()
+
+  return (
+    <DataView
+      fields={[
+        { label: 'Date', priority: 1 },
+        { label: 'Status', priority: 1 },
+        { label: 'Action', priority: 3 },
+        { label: 'Collateral state', priority: 1 },
+        { label: 'Amount', priority: 1 },
+      ]}
+      entries={stakingMovements}
+      renderEntry={({
+        amount,
+        createdAt,
+        actionState,
+        collateralState,
+        tokenDecimals,
+        disputableActionId,
+      }) => {
+        const actionAttributes = {
+          Scheduled: {
+            background: theme.infoSurface,
+            color: theme.tagIndicatorContent,
+            icon: <IconClock size="small" />,
+          },
+          Challenged: {
+            background: theme.warningSurface,
+            color: theme.warningSurfaceContent,
+            icon: <IconAttention size="small" />,
+          },
+          Completed: {
+            background: theme.positiveSurface,
+            color: theme.positiveSurfaceContent,
+            icon: <IconCheck size="small" />,
+          },
+          Cancelled: {
+            background: theme.surfaceUnder,
+            color: theme.contentSecondary,
+            icon: <IconCross size="small" />,
+          },
+        }
+        const amountAttributes = {
+          Locked: {
+            color: theme.surfaceOpened,
+            icon: (
+              <IconLock
+                size="small"
+                css={`
+                  margin-right: ${1 * GU}px;
+                `}
+              />
+            ),
+          },
+          Challenged: {
+            color: theme.surfaceOpened,
+            icon: (
+              <IconLock
+                size="small"
+                css={`
+                  margin-right: ${1 * GU}px;
+                `}
+              />
+            ),
+          },
```

Comment:
To avoid the duplication we can apply the icon size and margin at the point where we render, e.g.

```javascript
Locked: {
    color: theme.surfaceOpened,
    icon: IconLock,
},
```

```javascript
{icon && <Icon size="small" css={...}}
```